### ISSUE TITLE
Need to be able to see both inbound and outbound reports regarding other instances

### ISSUE THREAD

**Issue by quintessence** [mastodon/mastodon#31234]
### Pitch

Something that comes up frequently when "moderator relations go wrong" is reports sent/received and what was actioned. There should be a way for moderators to quickly see both what reports were filed against an instance but also what reports were filed _from_ that instance.

### Motivation

Recently we ran into a situation where another instance stated that they send us reports that were not actioned. We used the search tools available and could not find any tickets from that instance. Because the dashboard only reports tickets filed _against_ an instance, we aren't sure if the search tool works the same. Basically: we're not able to confirm or deny what they're saying at all. See (sanitized) screenshots.

![image](https://github.com/user-attachments/assets/8172f5ea-82c9-46c9-87e9-be57e1209607)

![image](https://github.com/user-attachments/assets/75b061dd-d911-4943-a305-7d17c5026d09)

The format of the URL for the search is:

```
https://hachyderm.io/admin/reports?by_target_domain=example.com
```

Depending on how the search function works, it's either searching both reports originating from that domain and also reports sent to that domain. I suspect it's only showing us outbound reports (our users to their server), of which there are still none, but that doesn't help answer the particular situation being posed.

**Comment by ThisIsMissEm** [comment]
It looks like we'd need to adjust the [`ReportFilter`](https://github.com/mastodon/mastodon/blob/2016c5d912f400ae98ee03ce269112de2f9ec62d/app/models/report_filter.rb) to include a `by_source_domain` option, which works similarly to `by_target_domain`, so something like:

```ruby
Report.where(account: Account.where(domain: value))
```

We'd also need to then change the handling of the search field to allow selecting between source domain and target domain.

**Comment by ThisIsMissEm** [comment]
I'm thinking UI something like this would be necessary?

<img width="869" alt="Screenshot 2024-07-31 at 23 41 24" src="https://github.com/user-attachments/assets/9a2e5e59-15b2-4d8f-a47a-b563683c7ef2">

**Comment by ThisIsMissEm** [comment]
For the instance details page, we likely need an entirely new metric, since at the moment `Admin::Metrics::Measure::InstanceReportsMeasure` focuses solely on reports about an instance, not reports received from an instance.

**Comment by quintessence** [comment]
A bit of scope creep for the original request, but adding on to cd24's comment: I think it would be a great UI improvement if the search results (when there are any) surface the data like action taken, any notes, etc. in a more digestible way. The current search works fine for viewing individual reports, but not for trying to find aggregate patterns, which important to track 1 ) harmful users over time 2 ) harmful instances over time.

**Comment by ThisIsMissEm** [comment]
I don't think we can surface this easily, but I think either using the API to retrieve the data in bulk or having a data export functionality would be the way to go for this.

The reason we cannot probably surface this easily comes down to database performance + loading too many related objects.

**Comment by ThisIsMissEm** [comment]
@cd24 just a small note: I can't actually filter by what action was taken on the report, because that information simply does not exist; we only record when and who took action, not what action was taken. (it only exists in the audit log and in the change to the record for either Account or Status)

**Comment by ThisIsMissEm** [comment]
I've started work on this in #31449
